lib/ssh_scan/database.rb

```diff
@@ -0,0 +1,30 @@
+# API Databases
+require 'ssh_scan/database/mongo'
+require 'ssh_scan/database/sqlite'
+
+module SSHScan
+  class DatabaseConfig
+    def initialize(opts = {})
+      @@db_type = if !opts || opts.empty?
+                   'sqlite'
+                 else
+                   opts['database']['type']
+                 end
+    end
+
+    def self.from_config_file
+      @db_path = File.join(Dir.pwd, '/config/database/database_config.yml')
+      db_opts = YAML.load_file(@db_path)
+      SSHScan::DatabaseConfig.new(db_opts)
+    end
+
+    def self.set_database
+      from_config_file
+      if @@db_type.eql? 'mongodb'
```

Comment:
This is because that is a class method and not an instance method.

http://www.railstips.org/blog/archives/2009/05/11/class-and-instance-methods-in-ruby/